src/notebook/example.md

```diff
@@ -0,0 +1,145 @@
+
+query = """
+SELECT
+  Station.nom_station,
+  Station.id_station_itinerance,
+  Amenageur.nom_amenageur as amenageur,
+  Localisation."coordonneesXY" as geom
+FROM
+  Station
+INNER JOIN Localisation ON Station.localisation_id = Localisation.id
+INNER JOIN Amenageur ON Station.amenageur_id = Amenageur.id
+"""
+
+
+with engine.connect() as conn:
+    # Query a PostgreSQL database using the PostGIS extension
+    stations = GeoDataFrame.from_postgis(query, conn)
+
+print(f"Loaded {len(stations.index)} stations")
+stations.sample(10)
+```
+
+```python
+# Display an interactive map of the stations
+stations.explore(column="amenageur")
+```
+
+### Example 2: explore operators distribution
+
+```python
+import pandas as pd
+
+query = """
+SELECT
+  Operateur.nom_operateur,
+  PointDeCharge.id_pdc_itinerance
+FROM
+  PointDeCharge
+  INNER JOIN Station ON PointDeCharge.station_id = Station.id
+  INNER JOIN Operateur ON Station.operateur_id = Operateur.id
+"""
+
+with engine.connect() as conn:
+    # Query a PostgreSQL database using the PostGIS extension
+    pdcs = pd.read_sql_query(query, conn)
+
+print(f"Loaded {len(pdcs.index)} points of charge")
+pdcs.sample(10)
+```
+
+```python
+import seaborn as sns
+
+# Render a barplot with the number of points of charge by operator
+sns.barplot(data=pdcs.value_counts("nom_operateur"))
+```
+
+## Write data to the database
+
+### Example 1: create a new table with calculated indicator
+
+In this example, we will calculate the number of points of charge per French department at a particular date/time (now) and store this stateful snapshot in the database.
+
+```python
+import uuid
+import pandas as pd
+
+# Get the city code insee for each point of charge
+query = """
+SELECT
+  Localisation.code_insee_commune
+FROM
+  PointDeCharge
+  INNER JOIN Station ON PointDeCharge.station_id = Station.id
+  INNER JOIN Localisation ON Station.localisation_id = Localisation.id
+"""
+with engine.connect() as conn:
+    # Query a PostgreSQL database using the PostGIS extension
+    codes_insee = pd.read_sql_query(query, conn)
+
+# Add a department column
+codes_insee["department"] = codes_insee["code_insee_commune"].map(lambda x: int(x[:2]) if x else None)
+
+# Calculate our indicator and add a timestamp to each department counts (row)
+indicator = codes_insee.value_counts("department").to_frame().reset_index()
+indicator["calculated_at"] = pd.Timestamp.now()
+
+# Set UUIDs as the index
+indicator["uuid"] = indicator.apply(lambda _: uuid.uuid4(), axis=1)
+indicator.set_index("uuid", inplace=True)
+
+# Explictly set the department column as integers
+indicator = indicator.astype({"department": "int32"})
+indicator
+```
+
+```python
+# Save the indicator to a (new) table
+indicator.to_sql("IDepartmentDynamic", engine, if_exists="append")
+```
+
+```python
+# Check inserted results
+query = 'SELECT * FROM "IDepartmentDynamic" WHERE department = 75'
+dpmts = pd.read_sql_query(query, engine) 
+dpmts
```

Comment:
```suggestion
paris = pd.read_sql_query(query, engine) 
paris
```

Reply:
The branch don't include seed-api -> i can't test with filled tables but i don't see any malfunction. 
The Metabase APIs are also usable.
-> we need to discuss options for implementing indicators under Metabase